lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,112 @@
+package handler
+
+import (
+	"context"
+	"log"
+
+	domainmodels "code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+)
+
+// Handler handles bets and event updates.
+type Handler struct {
+	betRepository BetRepository
+}
+
+// New creates and returns a new Handler.
+func New(betRepository BetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+// HandleBets handles bets.
+func (h *Handler) HandleBets(
+	ctx context.Context,
+	bets <-chan rabbitmqmodels.Bet,
+) <-chan rabbitmqmodels.BetCalculated {
+	resultingBets := make(chan rabbitmqmodels.BetCalculated)
+
+	go func() {
+		defer close(resultingBets)
+
+		for bet := range bets {
+			log.Println("Processing bet, betId:", bet.Id)
+
+			// Calculate the domain bet based on the incoming bet received.
+			domainBet := domainmodels.BetCalculated{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			_, exists, _ := h.betRepository.GetByID(ctx, domainBet.Id)
+			if !exists {
+				// Insert the domain bet into the repository.
+				err := h.betRepository.InsertBet(ctx, domainBet)
+				if err != nil {
+					log.Println("Failed to insert bet, error: ", err)
+					continue
+				}
+			}
```

Comment:
If it exists, the bet should be updated. Imagine if there was another microservice in this system, e.g. that service is updating `Payment` of the bet. Controller would pick up the change and produce it to bets queue, while Calculator would need to update that `Payment` in its local database.